src/compiler/scala/tools/nsc/util/package.scala

```diff
@@ -64,31 +64,33 @@ package object util {
     writer.toString()
   }
 
-  /** Generate a string using a routine that wants to write on a stream. */
+  /** Generate a string using a routine that wants to write on a writer. */
   def stringFromWriter(writer: PrintWriter => Unit): String = {
     val stringWriter = new StringWriter()
     val stream = new NewLinePrintWriter(stringWriter)
     writer(stream)
     stream.close()
     stringWriter.toString
   }
+  /** Generate a string using a routine that wants to write on a stream. */
   def stringFromStream(stream: OutputStream => Unit): String = {
+    val utf8 = java.nio.charset.StandardCharsets.UTF_8
     val bs = new ByteArrayOutputStream()
-    val ps = new PrintStream(bs)
+    val ps = new PrintStream(bs, /*autoflush=*/ false, utf8.name)    // use Charset directly in jdk10
     stream(ps)
     ps.close()
-    bs.toString()
+    bs.toString(utf8.name)
   }
-  def stackTraceString(ex: Throwable): String = stringFromWriter(ex printStackTrace _)
+  def stackTraceString(t: Throwable): String = stringFromWriter(t.printStackTrace(_))
 
   /** A one line string which contains the class of the exception, the
    *  message if any, and the first non-Predef location in the stack trace
    *  (to exclude assert, require, etc.)
    */
-  def stackTraceHeadString(ex: Throwable): String = {
```

Comment:
This tic has to do with avoiding modeline detection in vi. Maybe I'll contribute a scalafix for `ex:`.